nirc_ehr/resources/queries/study/housing.js

```diff
@@ -4,19 +4,66 @@ var prevDate;
 
 EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.BEFORE_INSERT, 'study', 'housing', function (helper, scriptErrors, row, oldRow) {
 
-    var isTransfer = prevAnimalId === row.Id;
+    if (helper.isETL()) {
+        var isTransfer = prevAnimalId === row.Id;
 
-    if (isTransfer) {
-        row.endDate = prevDate;
+        if (isTransfer) {
+            row.endDate = prevDate;
+        }
     }
 
-    if (row.endDate === 'undefined') {
-        console.log("end date not found for animal event - " + row.objectId);
+        if (row.endDate === 'undefined') {
+            console.log("end date not found for animal event - " + row.objectId);
+        }
+
+        prevAnimalId = row.Id;
+        prevDate = row.date;
+    });
+
+function onComplete(event, errors, helper){
+
+    // Similar to EHR housing trigger onComplete with minor differences to handle ETL
+    var idsToClose = [];
+    if (helper.isETL()) {
+        var boundaryRows = helper.getRows();
+        if (boundaryRows && boundaryRows.length > 0) {
+            boundaryRows = [boundaryRows[0]];  // Just need first boundary row for this extra step
+            for (var i = 0; i < boundaryRows.length; i++) {
+                if (EHR.Server.Security.getQCStateByLabel(boundaryRows[i].row.QCStateLabel).PublicData && boundaryRows[i].row.date) {
+                    boundaryRows[i].row.date.setHours(12);  // This just prevents warnings in EHR
+                    idsToClose.push({
+                        Id: boundaryRows[i].row.Id,
+                        date: EHR.Server.Utils.datetimeToString(boundaryRows[i].row.date),  //stringify to serialize properly
+                        objectid: boundaryRows[i].row.objectid
+                    });
+                }
+            }
+        }
     }
 
-    prevAnimalId = row.Id;
-    prevDate = row.date;
+    else if (!helper.isValidateOnly()) {
```

Comment:
This looks like identical code (aside from variable names and logging) to the block above. Can it be collapsed?

Reply:
Yes, I cleaned this up a bit.  Rushed this PR